tracer/src/Datadog.Trace/AppSec/BlockException.cs

```diff
@@ -5,13 +5,15 @@
 
 #nullable enable
 using System;
+using System.Collections.Generic;
 using Datadog.Trace.AppSec.Waf;
 using Datadog.Trace.ClrProfiler.CallTarget;
 
 namespace Datadog.Trace.AppSec
 {
     internal class BlockException : CallTargetBubbleUpException
     {
+        // This constructor is not referenced but it's used with reflection
```

Comment:
If you add this attribute, the IDE won't complain about this 😄 (Lucas taught me this the other day!)

```suggestion
        // This constructor is not referenced but it's used with reflection
        [Serializable]
```